File: apps/mobile/modules/native-module-demo/android/src/main/java/dev/schnau/studienbuch/localmodule/StudienbuchLocalModule.kt
package dev.schnau.studienbuch.localmodule

import android.os.Build
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class StudienbuchLocalModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("StudienbuchLocalModule")

    Constant("moduleKind") {
      "local"
    }

    Function("getModuleBoundary") {
      "apps/mobile/modules/native-module-demo"
    }

    Function("getPlatformSummary") {
      "Android ${Build.VERSION.RELEASE} (API ${Build.VERSION.SDK_INT}) via Kotlin local Expo module"
    }

    View(StudienbuchNativeBadgeView::class) {
      Prop("title") { view: StudienbuchNativeBadgeView, title: String ->
        view.title = title
      }

      Prop("accentColor") { view: StudienbuchNativeBadgeView, accentColor: String ->
        view.accentColor = accentColor
      }
    }
  }
}

File: apps/mobile/modules/native-module-demo/ios/StudienbuchLocalModule.swift
import ExpoModulesCore
import UIKit

public class StudienbuchLocalModule: Module {
  public func definition() -> ModuleDefinition {
    Name("StudienbuchLocalModule")

    Constant("moduleKind") {
      "local"
    }

    Function("getModuleBoundary") { () -> String in
      "apps/mobile/modules/native-module-demo"
    }

    Function("getPlatformSummary") { () -> String in
      let device = UIDevice.current
      return "\(device.systemName) \(device.systemVersion) via Swift local Expo module"
    }

    View(StudienbuchNativeBadgeView.self) {
      Prop("title") { (view, title: String) in
        view.title = title
      }

      Prop("accentColor") { (view, accentColor: String) in
        view.accentColor = accentColor
      }
    }
  }
}
